### DISCUSSION TITLE
Support ESM for sub-packages

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#4680] [Ideas]
Should ship ESM for sub-packages such as `@pnpm/find-workspace-packages`.

Need to make this change to each package and change the build script.

```
exports: {
    "types": "./src/index.js", // For TypeScript (needs .js instead of .ts even though that file doesn't actually exist)
    "import": "./lib/esm/index.js",
    "require": "./lib/cjs/index.js"
}
```

Would also need to check over each module to make sure they don't have any non-ESM compatible stuff like `dirname`.

# Reference

[node.js docs - Writing dual packages while avoiding or minimizing hazards](https://nodejs.org/api/packages.html#writing-dual-packages-while-avoiding-or-minimizing-hazards)

**Comment by zkochan** [comment]
I would rather just switch to ESM. But Jest doesn't work well with ESM yet.

https://github.com/pnpm/pnpm/pull/3139
